server/routes/index.ts

```diff
@@ -70,12 +70,14 @@ function createRouter(repositories: Repositories) {
             res.status(403).send(`You do not have permission to edit user id ${req.targetUser!.id}`);
             return next();
         }
-        req.user.updateFromData(req.body);
-        users.save(req.user)
-            .then(() => {
-                res.redirect(req.targetUser!.url);
-                next();
+        req.targetUser!.updateFromData(req.body);
+        if (req.targetUser) {
```

Comment:
You are calling a method on targetUser above. How could it possible be null here?

`!.` is not an operator. `req.targetUser!` asserts that `req.targetUser` is not null

Reply:
error TS2345: Argument of type 'User | undefined' is not assignable to parameter of type 'DeepPartial<User>'.

Reply:
req.targetUser! asserts that req.targetUser is not null

Reply:
go ahead and compile it without the if statement then

Reply:
![image](https://user-images.githubusercontent.com/4969631/32574380-4aa9e65c-c48e-11e7-9a62-b42a71ef43df.png)
Works fine for me. When I assert that `req.targetUser` is not null

Reply:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

Reply:
Sorry if I was unclear, but it does the assertion at the expression level. So you have to do it every time you use a variable that has a type `undefined | T` or `null | T`

Reply:
Oh, i'm just now seeing the assertion you added to save. Apologies for my confusion